src/trigger.F90

```diff
@@ -0,0 +1,541 @@
+                    trigger % variance = std_dev ** 2 
+                  end if
+                  if (trigger % std_dev < std_dev) then 
+                    trigger % std_dev = std_dev
+                  end if
+                  if (trigger % rel_err < rel_err) then 
+                    trigger % rel_err = rel_err
+                  end if
+
+                end select
+ 
+                select case (t % triggers(s) % type)        
+                case(VARIANCE) 
+                  uncertainty = trigger % variance
+                case(STANDARD_DEVIATION)
+                  uncertainty = trigger % std_dev
+                case default
+                  uncertainty = trigger % rel_err
+                end select
+              
+                if (uncertainty > t % triggers(s) % threshold) then
+                  satisfy_triggers = .false.
+
+                  if (t % triggers(s) % type == VARIANCE) then
+                    ratio = sqrt(uncertainty / t % triggers(s) % threshold)
+                  else
+                    ratio = uncertainty / t % triggers(s) % threshold
+                  end if 
+                  
+                  if (trig_dist % max_ratio < ratio) then
+                    trig_dist % max_ratio = ratio
+                    trig_dist % name  = t % triggers(s) % score_name
+                    trig_dist % tally_id = t % id
+                  end if
+                end if
+              end do NUCLIDE_LOOP
+              if (t % n_filters == 0) exit FILTER_LOOP
+            end do FILTER_LOOP
+          end if
+        end do TRIGGER_LOOP
+      end do TALLY_LOOP
+    end if
+  end subroutine check_tally_triggers
+
+
+!===============================================================================
+! COMPUTE_TALLY_CURRENT computes the current for a surface current tally with
+! precision trigger(s).
+!===============================================================================
+ 
+ subroutine compute_tally_current(t, trigger)
+
+    integer :: i                    ! mesh index for x
+    integer :: j                    ! mesh index for y
+    integer :: k                    ! mesh index for z
+    integer :: l                    ! index for energy
+    integer :: i_filter_mesh        ! index for mesh filter
+    integer :: i_filter_ein         ! index for incoming energy filter
+    integer :: i_filter_surf        ! index for surface filter
+    integer :: n                    ! number of incoming energy bins
+    integer :: filter_index         ! index in results array for filters
+    logical :: print_ebin           ! should incoming energy bin be displayed?
+    real(8) :: rel_err  = 0.0         ! temporary relative error of result
+    real(8) :: std_dev  = 0.0         ! temporary standard deviration of result
+    type(TallyObject), pointer :: t   ! tally pointer
+    type(TriggerObject)   :: trigger  ! tally trigger
+    type(StructuredMesh), pointer :: m => null()
```

Comment:
You can remove the `=> null()` initialization. That ensures the pointer is allocated on the stack.